locallib.php

```diff
@@ -167,6 +169,28 @@ function local_boostnavigation_build_custom_nodes($customnodes, navigation_node
                             // so that the custom node is displayed.
                             $nodevisible &= local_boostnavigation_user_has_role_on_page($USER->id, $setting);
 
+                            break;
+                        // Check for the optional sixth parameter: system role filter.
+                        case 5:
+                            // Only proceed if some role is entered here. This parameter is optional.
+                            // If no system role shortnames are given, the node will not be added to the navigation by default.
+                            // Otherwise, it is checked whether the user has any of the provided roles,
+                            // so that the custom node is displayed.
+                            if (local_boostnavigation_user_has_role_on_system($USER->id, $setting) && $nodelanguage) {
+                                $nodevisible = true;
+                            }
+
+                            break;
+                        // Check for the optional seventh parameter: admin filter.
```

Comment:
As written above: I would be happy to see this dedicated parameter included in the sixth parameter